### PR TITLE
Auto image change 

### PR DESCRIPTION
**Pull request by lodyczekoladowe**

### CODE REVIEW

**Review by JulianWielga — CHANGES_REQUESTED** [review]

File: designer/client/cypress/tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "compilerOptions": {
         "target": "es5",
-        "lib": ["es5", "dom"],
+        "lib": ["es5", "dom", "ES2021.String"],
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/tsconfig.json:4]: no

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        postSchemaToRegistry();
+    });
+
+    afterEach(() => {
+        cy.deleteAllTestProcesses({ filter: seed, force: true });
+    });
+
+    it("should display - basic components - variable", () => {
+        cy.visitNewProcess(seed, "docsBasicComponentsVariable#0");
+        cy.get('[title="layout"] > .ToolbarButton--icon--w9M5n').click();
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:19]: okey

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        cy.get('[title="Name"]').click();
+        imageTake();
+    });
+
+    it("should display - basic components - mapmariable", () => {
+        cy.visitNewProcess(seed, "docsBasicComponentsMapVariable#0");
+        cy.get('[title="layout"] > .ToolbarButton--icon--w9M5n').click();
+
+        cy.get('[model-id="node label goes here"]').dblclick();
+        cy.get('[title="Name"]').click();
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:37]: no, the point is to make the cursor stop flashing

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+        cy.get('[model-id="kafka"]').dblclick();
+        cy.get('[title="Name"]').click();
+        imageTake();
+    });
+});
+
+function imageTake(padding = false) {
+    const basePath = "/home/fgl/Git/nussknacker/docs/scenarios_authoring/img/imagesComparing/";
+    cy.get(padding ? ".joint-layers" : '[data-testid="window-frame"]').matchImage({
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:152]: okey

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        cy.get('[title="layout"] > .ToolbarButton--icon--w9M5n').click();
+
+        cy.get('[model-id="kafka"]').dblclick();
+        cy.get('[title="Name"]').click();
+        imageTake();
+    });
+});
+
+function imageTake(padding = false) {
+    const basePath = "/home/fgl/Git/nussknacker/docs/scenarios_authoring/img/imagesComparing/";
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:151]: i dont know how to fix that

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        cy.visitNewProcess(seed, "dockKafkaInput#0");
+        cy.get('[title="layout"] > .ToolbarButton--icon--w9M5n').click();
+
+        cy.get('[model-id="kafka"]').dblclick();
+        cy.get('[title="Name"]').click();
+        imageTake();
+    });
+});
+
+function imageTake(padding = false) {
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:150]: okey

File: designer/client/cypress/e2e/imagesComparing.cy.ts [hunk 7]
```diff
@@ -0,0 +1,189 @@
+describe("Images Comparing", () => {
+    const seed = "imagesComparing";
+
+    before(() => {
+        cy.deleteAllTestProcesses({ filter: seed, force: true });
+    });
+
+    beforeEach(() => {
+        cy.viewport(1920, 1080);
```
[the remark this answers is not in the document]
Comment (lodyczekoladowe) [reply designer/client/cypress/e2e/imagesComparing.cy.ts:9]: however it has to be

File: CONTRIBUTING.md
Comment (lodyczekoladowe) [inline CONTRIBUTING.md:1]: @ZbyszekMM Is it okey?

**Review by JulianWielga — APPROVED** [review]

**Review by JulianWielga — CHANGES_REQUESTED** [review]

**Review by ZbyszekMM — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -65,8 +65,47 @@ When writing documentation please follow these instructions:
...
     [Components](../about/GLOSSARY#component)
     ```
+### Automatic Image Updates in Documentation
 
-## Working with code
+To automate image updates in your documentation, follow these steps:
+
+1. **Create a Cypress Test**: Begin by adding a new Cypress test in the `designer/client/cypress/e2e/autoImageChangeDocs.cy.ts` file. In this test, utilize one of the image capture functions available in your code.
+
+2. **Image Storage**: All captured images are stored in the `docs/autoImageChangeDocs` folder. These images are named according to a specific convention:
```
Comment (ZbyszekMM) [inline CONTRIBUTING.md:74]: specific --> following

**Review by JulianWielga — APPROVED** [review]

**Pull request merged**